doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,951 @@
+dpu_midplane_link_state: up refers to the pcie link between the NPU and DPU is operational. This will be updated by the switch pcied.
+
+dpu_booted_state: up refers to SONiC booted on DPU
+
+dpu_control_plane_state: up  refers to all the containers are up and the interfaces are up and dpu is ready to take SDN policy update
+
+dpu_data_plane_state: up  refers to configuration downloaded, the pipeline stages are up and DPU hardware (port/ASIC) is ready to take traffic
+
+```
+    ChassisStateDB Schema for DPU_STATE
+
+    Table: “DPU_STATE”
+
+    SCHEMA
+    key:  DPU_STATE|DPU0
+
+        "id": "1",
+        ”dpu_midplane_link_state”: "up"
+        “dpu_midplane_link_time": "timestamp",
+        "dpu_midplane_link_reason": "up_down_related string",
+        ”dpu_booted_state": "up",
+        "dpu_booted_time": "timestamp",
+        "dpu_booted_reason": "up_down_related string",
+        ”dpu_control_plane_state": "up"
+        ”dpu_control_plane_time": ”timestamp",
+        ”dpu_control_plane_reason": ”containers restarting",
+        ”dpu_data_plane_state": ”down",
+        ”dpu_data_plane_time": ”timestamp",
+        ”dpu_data_plane_reason": ”Pipeline failure",
+```
+
+3. Each DPU has to store the health data in its local DB and should provide it to the switch.
+* When the "show system-health ..." CLI is executed on the switch, the "UserDefinedChecker" class will collect the data and feed it to the CLI. It is up to the platform on how this is done.  However, for faster access store it in the switch chasisStateDB.
+* The DPU is a complex hardware, to facilitate debug, a consistent way of storing and accessing the health record of the DPUs is critical in a multi vendor scenario even though it is a platform specific implementation.
+* Both switch and the DPUs will follow to the [SONiC system health monitor HLD](https://github.com/sonic-net/SONiC/blob/ce313db92a694e007a6c5332ce3267ac158290f6/doc/system_health_monitoring/system-health-HLD.md)
+* Refer to section 3.4.5 for "show system-health .." CLIs
+
+##### 3.1.5.2 ModuleBase class new APIs
+The DPU ID is used only for indexing purpose.
+
+get_dpu_id(self):
+```
+    Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+
+    Returns:
+        An integer, indicating the DPU ID. DPU0 returns 1, DPUX returns X+1
+        Returns '0' on switch module
+```
+#### Get DPU reboot cause
+def get_reboot_cause(self):
+```
+        Retrieves the cause of the previous reboot of the DPU module
+
+        Returns:
+            A tuple (string, string) where the first element is a string containing the cause of the previous reboot. This string must be one of the predefined strings in this class. If the first string is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be used to pass a description of the reboot cause.
+```
+
+#### DPU_STATE Use Case
+* High Availability, Load Balancing, Debug, error recovery (reset, power cycle) and fault management logics will use this API
+
+get_state_info(self):
```

Comment:
The understanding is that the HA logic may use this.  However, how and when will be defined in the HA document.  If needed at that time will update this section.  Also, the there was minor issue that mentioned that this is used in the DPU instead of SWITCH, which is fixed now.  The purpose of this is to keep this as a fall back option or even as the main option in case of any BFD implementation issues.